### PR TITLE
Adding types/memoize with TTL + use it in connectors

### PR DESCRIPTION
**Pull request by lasryaric**
- Removed `lodash.memoize` from connectors/package.json
- Created our own `@dust-tt/types` `memoize` function, with TTL support.
- Replaced all calls to lodash memoize with our own memoize and a 10 minutes TTL.
- TTL is from last call to the memoize function.

### CODE REVIEW

**Review by spolu — APPROVED** [review]

File: connectors/src/connectors/slack/temporal/activities.ts [hunk 1]
```diff
@@ -1,4 +1,8 @@
-import { CoreAPIDataSourceDocumentSection, ModelId } from "@dust-tt/types";
+import {
+  CoreAPIDataSourceDocumentSection,
+  memoize,
```
Comment (fontanierh) [inline connectors/src/connectors/slack/temporal/activities.ts:3]: the package naming is already outdated 😓

Comment (lasryaric) [reply connectors/src/connectors/slack/temporal/activities.ts:3]: @fontanierh we can change it anytime, no big deal.

**Review by fontanierh — APPROVED** [review]
Looks good to me, interesting solution (I guess it's fine to create a lot of timeouts ?)

**Comment by lasryaric** [comment]
@fontanierh yes, you always clean up the previous one before creating the new one so it's all good.

**Pull request merged**